src/Gateway/Form/Type/AbstractGatewayConfigurationType.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace PayPlug\SyliusPayPlugPlugin\Gateway\Form\Type;
+
+use Sylius\Bundle\PayumBundle\Model\GatewayConfigInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class AbstractGatewayConfigurationType extends AbstractType
+{
+    /** @var \Symfony\Contracts\Translation\TranslatorInterface */
+    protected $translator;
+
+    /** @var FlashBagInterface */
+    protected $flashBag;
+    /**
+     * @var \Sylius\Component\Resource\Repository\RepositoryInterface
+     */
+    private $gatewayConfigRepository;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        FlashBagInterface $flashBag,
+        RepositoryInterface $gatewayConfigRepository
+    ) {
+        $this->translator = $translator;
+        $this->flashBag = $flashBag;
+        $this->gatewayConfigRepository = $gatewayConfigRepository;
+    }
+
+    protected function canBeCreated(string $factoryName): bool
+    {
+        $alreadyExists = $this->gatewayConfigRepository->findOneBy(['factoryName' => $factoryName]);
+
+        if (!$alreadyExists instanceof GatewayConfigInterface) {
+            return true;
+        }
+
+        return false;
```

Comment:
On peut factoriser comme cela ?
```suggestion
        return !$alreadyExists instanceof GatewayConfigInterface
```